### ISSUE TITLE
I only get around 400pps when using -i1

### ISSUE THREAD

**Issue by k79e** [schweikert/fping#159]
IDK is it normal?
But when I use -i0 it got faster. 
Then it dropped. I don't know why but -i1 can never be faster. 
When I using -i0 the speed looks dynamic. (Maybe it's because I used all cpu? But with -i1 it never 100% cpu)

I used it on lan to test and found that -i1 can't even up to 1000pps!!!
It looks around 300pings/s
Then I changed to a value which smaller than 1 and speed become veryfast. 

2019-10-05 22:28:19
Well I found that pps dropped to around 300 but fping still using 100% cpu!!!! Maybe it becomes stucks?? when I use -i0?
But I when I run another instence at the same time to ping a lan ip it can faster than 1000pps

Hey didn't this tool is async ping so the sending rate should be fixed?

**Comment by Jimmy-Z** [comment]
I'm not the developer but I'm a bit curious about this to so I did some test myself
```
# time fping -c 1000 -t 100 -i 1 -p 1 -q 192.168.0.1
192.168.0.1 : xmt/rcv/%loss = 1000/1000/0%, min/avg/max = 0.20/0.70/2.12

real    0m1.070s
user    0m0.000s
sys     0m0.080s
```
So doesn't seem to be true here. 192.168.0.1 is another vm but on the same host.

```
# time fping -c 1000 -t 100 -i 0 -p 0 -q 192.168.0.1
192.168.0.1 : xmt/rcv/%loss = 1000/652/34%, min/avg/max = 7.26/26.0/43.0

real    0m0.106s
user    0m0.001s
sys     0m0.007s
```
I never knew you could `-i 0` (man page says minimum 1), I'm reaching 9.4kpps AFAICT, but that 34% loss rate is something alarming, that's a pfsense gateway, maybe it's limiting echo rate.

```
# time fping -c 1000 -t 100 -i 0 -p 0 -q 127.0.0.1
127.0.0.1 : xmt/rcv/%loss = 1000/1000/0%, min/avg/max = 0.00/0.00/0.29

real    0m0.014s
user    0m0.000s
sys     0m0.013s
```
This is reaching 71kpps.

**Comment by kxmp** [comment]
Mine
 time fping -c5000 -i1 -p1 127.0.0.1 -q
127.0.0.1 : xmt/rcv/%loss = 5000/5000/0%, min/avg/max = 0.01/0.04/0.23

real    0m9.880s
user    0m0.000s
sys     0m0.636s

**Comment by schweikert** [comment]
I don't understand what problem you are describing. -i1 means that fping will verify that no more than 1 ping is sent every 1 ms. That explains why you can't get more than 1000pps with that.

**Comment by Akalanka1337** [comment]
What type of packets are received ?
fping -e -c 3 192.168.112.168 

?
